ntp-udp/src/lib.rs

```diff
@@ -98,28 +98,91 @@ impl AsRef<std::net::UdpSocket> for UdpSocket {
     }
 }
 
-fn init_socket(socket: &std::net::UdpSocket) -> io::Result<()> {
-    let fd = socket.as_raw_fd();
-    let enable_ts: libc::c_int = 1;
+fn set_timestamping_options(udp_socket: &std::net::UdpSocket) -> io::Result<()> {
+    let fd = udp_socket.as_raw_fd();
+
+    // our options:
+    //  - we want software timestamps to be reported,
+    //  - we want receive software timestamps
+    let options = libc::SOF_TIMESTAMPING_SOFTWARE | libc::SOF_TIMESTAMPING_RX_SOFTWARE;
+
     unsafe {
-        libc::setsockopt(
+        cerr(libc::setsockopt(
             fd,
             libc::SOL_SOCKET,
-            libc::SO_TIMESTAMPNS,
-            &enable_ts as *const _ as *const libc::c_void,
+            libc::SO_TIMESTAMPING,
```

Comment:
now using `SO_TIMESTAMPING` because it will allow us to do send timestamping in the future